content/trust-management/trust-authority.md

```diff
@@ -60,23 +56,163 @@ its list of public signature verification keys by removing the compromised key
+        "IlRMeOPX2e0MurIyfWEucYBRVOEEUMrOHqn/8mLqMjA"
+    ],
+    "website": "https://www.somengo.org",
+    "description": {
+        "en": "We do activism for freedom of the press.",
+        "fr": "Nous sommes des activistes en faveur de la liberté de la presse."
+    },
+    "logo": "mxc://weu.informo.network/AtEuTuVSeIlZQgjEzjGyMDtG",
+    "trusted": {
+        "@acmenews:example.com": {
+            "signature": "0a1df56f1c3ab5b1",
+            "type": "source"
+        },
+        "@someothersource:example2.com": {
+            "signature": "daiRanaiy1be7pe",
+            "type": "source"
+        }
+    },
+    "blacklist": {
+        "@sadsource:example.com": {
+            "reason_code": "B_COMPROMISED",
+            "since": "!someEvent:example.com"
+        },
+        "@badsource:example.com": {
+            "reason_code": "B_MISINFORMATION",
+            "since": "",
+            "additional_info": {
+                "en": "This source actually belongs to the Tomainian government which uses it for propaganda.",
+                "fr": "Cette source appartient au gouvernement Tomanien qui l'utilise à des fins de propagande."
+            }
+        }
+    }
+}
+```
+
+## Blacklisting
+
+A trust authority **can** list sources and trust authorities that it explicitly
+blacklists for being compromised or ethical reasons. The trust authority
+**must** specify a reason for blacklisting a source or TA, which consists in a
+defined reason code that client implementations **can** use in order to warn
+users about a specific source. The trust authority **can** also provide
+additional information to explain the addition to the blacklist, which, if
+provided, **must** be contained in a custom localised string. Client
+implementations **can** also use this string to provide the users with more
+information on why a specific source was blacklisted by the trust authority.
+
+### Blacklist reason codes
+
+As defined above, a trust authority **must** provide a reason for blacklisting a
+source or another trust authority. This **must** be done using at least a reason
+code, which **must** be one of the following:
+
+|        Code        |            Meaning             |
+| ------------------ | ------------------------------ |
+| `B_COMPROMISED`    | One of the public keys of the source or trust authority has been compromised. |
+| `B_MISINFORMATION` | If the blacklisted entity is a source, it has been publishing false or unverified information intentionnally. If it's a trust authority, it has been certifying sources publishing false or unverified information as trustworthy intentionnally. |
+| `B_ABANDONED`      | The source or trust authority has ceased its activity and/or to publish articles through this federation, therefore the entity isn't used anymore and very unlikely to ever be used again. Blacklisting it then prevents someone else from getting their hands on the entity's keys and tokens and impersonate its former owner. |
+
```

Comment:
Why are the reason code prefixed with `B_` ?

Reply:
It stands for `BLACKLIST`. We might want to add predefined codes for other things in the future so it might make things easier to prefix them with the feature they relate to in case we see it mentioned in the wild or a conversation lacking enough context, just like Matrix's error codes use the form `E_SOME_ERROR`.

Reply:
right, good idea :+1: